include/swift/AST/EvaluatorDependencies.h

```diff
@@ -180,64 +173,25 @@ struct DependencyRecorder {
   friend DependencyCollector;
 
 private:
-  /// A stack of dependency sources in the order they were evaluated.
-  llvm::SmallVector<evaluator::DependencySource, 8> dependencySources;
-  llvm::DenseMap<SourceFile *, DependencyCollector::ReferenceSet>
+  llvm::DenseMap<SourceFile *,
+                 llvm::DenseSet<DependencyCollector::Reference,
+                                DependencyCollector::Reference::Info>>
       fileReferences;
-  llvm::DenseMap<AnyRequest, DependencyCollector::ReferenceSet>
+  llvm::DenseMap<AnyRequest, std::vector<DependencyCollector::Reference>>
       requestReferences;
-  bool isRecording;
```

Comment:
Please don’t delete the re-entrancy check entirely. It has caught a lot of bad bugs

Reply:
Good point, I'll add that back.